tsdb/tsm1/engine_schema.go

```diff
@@ -358,6 +358,12 @@ func (e *Engine) tagKeysNoPredicate(ctx context.Context, orgID, bucketID influxd
 	var stats cursors.CursorStats
 	var canceled bool
 
+	// NOTE(bbj): Adding padding in case additional files are added between files
+	// instantiation & the ForEachFile(). The 'files' reference needs to be stable
+	// so defer can clean it up at the end.
+	files := make(unrefs, 0, e.FileStore.Count()+10)
+	defer files.Unref()
```

Comment:
You can capture `files` in a closure, it won't matter if it grows:

https://play.golang.org/p/xYwL9_OT07S

Reply:
Ah, I like that better. 👍